api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -394,34 +394,22 @@ public boolean isSecretAnswer(User u, String answer) {
 	 * @param new user that has privileges
 	 */
 	private void checkPrivileges(User user) {
-		Collection<Role> roles = user.getAllRoles();
-		List<String> requiredPrivs = new ArrayList<>();
-		
-		for (Role r : roles) {
-			if (r.getRole().equals(RoleConstants.SUPERUSER)
-			        && !Context.hasPrivilege(PrivilegeConstants.ASSIGN_SYSTEM_DEVELOPER_ROLE)) {
-				throw new APIException("User.you.must.have.role", new Object[] { RoleConstants.SUPERUSER });
-			}
-			if (r.getPrivileges() != null) {
-				for (Privilege p : r.getPrivileges()) {
-					if (!Context.hasPrivilege(p.getPrivilege())) {
-						requiredPrivs.add(p.getPrivilege());
-					}
-				}
-			}
-		}
-		
+		List<String> requiredPrivs = user.getAllRoles().stream().peek(this::checkSuperUserPrivilege)
+				.map(Role::getPrivileges).filter(Objects::nonNull).flatMap(Collection::stream)
+				.map(Privilege::getPrivilege).filter(p -> !Context.hasPrivilege(p)).collect(Collectors.toList());
 		if (requiredPrivs.size() == 1) {
 			throw new APIException("User.you.must.have.privilege", new Object[] { requiredPrivs.get(0) });
 		} else if (requiredPrivs.size() > 1) {
-			StringBuilder txt = new StringBuilder("You must have the following privileges in order to assign them: ");
-			for (String s : requiredPrivs) {
-				txt.append(s).append(", ");
-			}
-			throw new APIException(txt.substring(0, txt.length() - 2));
+			throw new APIException(String.format(ERROR_FORMAT, String.join(", ", requiredPrivs)));
+		}
+		}
+
+	private void checkSuperUserPrivilege(Role r) {
+		if (r.getRole().equals(RoleConstants.SUPERUSER)
+				&& !Context.hasPrivilege(PrivilegeConstants.ASSIGN_SYSTEM_DEVELOPER_ROLE)) {
+			throw new APIException("User.you.must.have.role", new Object[] { RoleConstants.SUPERUSER });
+		}
 		}
```

Comment:
This should only be indented one level

Reply:
Are you meaning i format this code?

Reply:
Yes

Reply:
I have finished formatting it thanks @ibacher  , anything else

Reply:
I was suggesting that we leave the next  method as it is since it can be compared with method checkSuperUser. Tried to use Optional.ofNullable but still there are some Optional errors